pkg/devfile/parse_test.go

```diff
@@ -751,12 +775,38 @@ spec:
 				}
 				return nil
 			},
+			wantError: false,
+		},
+		{
+			name: "with unsupported schema version",
+			args: args{
+				args: parser.ParserArgs{
+					ExternalVariables: map[string]string{
+						"PARAMS": "baz",
+					},
+					Data: []byte(devfileContentWithUnsupportedSchema),
+				},
+			},
+			wantCommandLine: "./main baz",
+			wantVariables: map[string]string{
+				"PARAMS": "baz",
+			},
+			wantVarWarning: variables.VariableWarning{
+				Commands:        map[string][]string{},
+				Components:      map[string][]string{},
+				Projects:        map[string][]string{},
+				StarterProjects: map[string][]string{},
+			},
+			wantError: true,
 		},
 	}
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
 			gotD, gotVarWarning, err := ParseDevfileAndValidate(tt.args.args)
-			if err != nil {
+			if err != nil && tt.wantError {
+				t.Log("Correctly identified unsupported schema version")
+				return
+			} else if err != nil {
 				t.Errorf("ParseDevfileAndValidate() error = %v, wantErr nil", err)
```

Comment:
Not part of your changes but just a nit pick correction to the current source code:
```suggestion
				t.Errorf("ParseDevfileAndValidate() error = %v, wantErr false", err)
```

Or maybe with my other suggestion:
```suggestion
				t.Errorf("ParseDevfileAndValidate() error = %v, wantErrStr = %v", err, tt.wantErrStr)
```

Reply:
If using `wantErrStr` maybe typed as `*string` would show `nil` in cases where it is unset.